### PR TITLE
G2 see hps attending

### PR DESCRIPTION
**Pull request by jdselby24**

### CODE REVIEW

File: public/css/genericStyles.css [hunk 1]
```diff
@@ -2,6 +2,10 @@
     box-sizing: border-box;
 }
 
+.bold {
```
Comment (charliecog) [inline public/css/genericStyles.css]: not good naming - its only used on a couple of hiring partner names so make the class more semantically named

File: public/js/eventsPage.js [hunk 2]
```diff
@@ -92,18 +94,53 @@ function displayEventsHandler(events) {
...
-        eventGenerator(event)
+        eventGenerator(event, hiringPartners)
+        .then((event) => {
+            let data = {
+                event_id: event.id
+            }
+
+            let hiringPartnersDiv = document.querySelector(`.hiring-partners[data-eventId='${event.id}']`)
+
+            fetch('./api/getHpsByEventId', {
```
Comment (charliecog) [inline public/js/eventsPage.js]: this can be put into a named function so its reusable and makes code less gross

File: public/js/eventsPage.js [hunk 3]
```diff
@@ -92,18 +94,53 @@ function displayEventsHandler(events) {
...
+                    'Content-Type': 'application/json',
+                },
+                method: 'post',
+                body: JSON.stringify(data)
+
+        })
+        .then(response => response.json())
+        .then(response => {
+            if(response.length != 0) {
+                hiringPartnersDiv.innerHTML += `<h4>Attending hiring partners</h4>`
```
Comment (charliecog) [inline public/js/eventsPage.js]: these should be aggregated into a large string before being output all at once - at the moment if an error happens half way through then only part of the data will be output

File: public/js/eventsPage.js [hunk 4]
```diff
@@ -92,18 +94,53 @@ function displayEventsHandler(events) {
...
+})
 }
 
 /**
  * Outputs HTML elements with event details
  *
  * @param events an object which contains information about an event
  */
-async function eventGenerator(event) {
+
```
Comment (charliecog) [inline public/js/eventsPage.js]: why extra line break

File: src/Classes/Controllers/GetHiringPartnersByIdController.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     * Invoke function to get all hiring partners associated with an event and returns JSON
+     *
+     * @param Request $request
+     *
+     * @param Response $response
+     *
+     * @param $args
```
Comment (charliecog) [inline src/Classes/Controllers/GetHiringPartnersByIdController.php]: doc blocks arent done properly

File: src/Classes/Factories/GetHiringPartnersByIdControllerFactory.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+use Portal\Controllers\GetHiringPartnersByIdController;
+use Psr\Container\ContainerInterface;
+
+class GetHiringPartnersByIdControllerFactory
+{
+
+    /**
+     * Invoke function to create new GetHiringPartnersByIdControllers
+     *
+     * @param ContainerInterface $container
```
Comment (charliecog) [inline src/Classes/Factories/GetHiringPartnersByIdControllerFactory.php]: doc block

File: src/Classes/Models/EventModel.php [hunk 7]
```diff
@@ -111,7 +110,16 @@ public function addHPToEvent(int $hiringPartner, int $event, $attendees = null):
...
 
-    public function checklinkHP(int $hiringPartner, int $event): bool
+    /**
+     * checks that the hiring partner has successfully been linked to the event in the database
+     *
+     * @param int $hiringPartner hiring partner id
+     *
+     * @param int $event event id
+     *
+     * @return bool
```
Comment (charliecog) [inline src/Classes/Models/EventModel.php]: doc block

File: src/settings.php [hunk 8]
```diff
@@ -17,7 +17,7 @@
         ],
 
         'db' => [
-            'host' => 'mysql:host=db;',
+            'host' => 'mysql:host=127.0.0.1;',
```
Comment (charliecog) [inline src/settings.php]: dont commit this

File: public/js/eventsPage.js (hunk 4 above)
Comment (sudokufan) [reply public/js/eventsPage.js]: on it

File: src/settings.php (hunk 8 above)
Comment (bentleybiscuit) [reply src/settings.php]: ok

File: public/js/eventsPage.js (hunk 2 above)
Comment (cwoodwarduk) [reply public/js/eventsPage.js]: On this

File: public/js/eventsPage.js (hunk 4 above)
Comment (sudokufan) [reply public/js/eventsPage.js]: done

File: src/settings.php (hunk 8 above)
Comment (sudokufan) [reply src/settings.php]: done

File: public/js/eventsPage.js (hunk 2 above)
Comment (cwoodwarduk) [reply public/js/eventsPage.js]: Done!

File: src/Classes/Controllers/GetHiringPartnersByIdController.php (hunk 5 above)
Comment (cwoodwarduk) [reply src/Classes/Controllers/GetHiringPartnersByIdController.php]: Done.

File: src/Classes/Factories/GetHiringPartnersByIdControllerFactory.php (hunk 6 above)
Comment (cwoodwarduk) [reply src/Classes/Factories/GetHiringPartnersByIdControllerFactory.php]: Done.

File: src/Classes/Models/EventModel.php (hunk 7 above)
Comment (cwoodwarduk) [reply src/Classes/Models/EventModel.php]: Done

File: public/css/genericStyles.css [hunk 9]
```diff
@@ -2,6 +2,10 @@
     box-sizing: border-box;
 }
 
+.bold-text-hp {
```
Comment (charliecog) [inline public/css/genericStyles.css]: .hiring-partner span 
 would do the same

Comment (jdselby24) [inline public/css/genericStyles.css]:
```suggestion
.hiring-partner span {
```

**Pull request merged**